### PR TITLE
Add ability to connect and trigger events when Usage Log is triggered

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
See the details listed on [this linked PR](https://github.com/BHoM/BHoMAnalytics_Toolkit/pull/35).

### CODE REVIEW

File: UI_oM/TriggerLogUsageArgs.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.oM.UI
```
Comment (alexissantella) [inline UI_oM/TriggerLogUsageArgs.cs:7]:
```suggestion
namespace BH.oM.UI, IObject
```

File: UI_oM/TriggerLogUsageArgs.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+{
+    public class TriggerLogUsageArgs : EventArgs
+    {
+        public virtual object SelectedItem { get; set; }
+        public virtual string CallerName { get; set; }
+        public virtual string UIName { get; set; }
+        public virtual string UIVersion { get; set; }
+        public virtual Guid ComponentID { get; set; }
```
Comment (alexissantella) [inline UI_oM/TriggerLogUsageArgs.cs:15]: could we use some defaults?

**Review by alexissantella — CHANGES_REQUESTED** [review]
Requesting changes based on code review.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot fix copyright headers ref. 7921146293

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Review by travispotterBH — APPROVED** [review]
https://github.com/BHoM/BHoMAnalytics_Toolkit/pull/35#pullrequestreview-1084342381

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Review by alexissantella — APPROVED** [review]
Approved based on reviews of depending PRs. 
See: https://github.com/BHoM/BHoMAnalytics_Toolkit/pull/35

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**